src/codewhisperer/util/runtimeLanguageContext.ts

```diff
@@ -47,59 +68,78 @@ export class RuntimeLanguageContext {
             sh: 'shell',
             shellscript: 'shell',
             sql: 'sql',
+            shell: 'shell',
+            jsx: 'jsx',
+            tsx: 'tsx',
+            plaintext: 'plaintext',
         })
-        this.supportedLanguageExtensionMap = createConstantMap<CodeWhispererConstants.SupportedLanguage, string>({
+        this.supportedLanguageExtensionMap = createConstantMap<CodewhispererLanguage, string>({
             java: 'java',
             python: 'py',
-            javascriptreact: 'jsx',
+            jsx: 'jsx',
             javascript: 'js',
             typescript: 'ts',
-            typescriptreact: 'tsx',
+            tsx: 'tsx',
             csharp: 'cs',
             c: 'c',
-            c_cpp: 'cpp',
             cpp: 'cpp',
             go: 'go',
             kotlin: 'kt',
             php: 'php',
             ruby: 'rb',
             rust: 'rs',
             scala: 'scala',
-            sh: 'sh',
-            shellscript: 'sh',
+            shell: 'sh',
             sql: 'sql',
+            plaintext: 'txt',
         })
+    }
 
-        const values = Array.from(this.supportedLanguageMap.values())
-        const keys = Array.from(this.supportedLanguageMap.keys())
-        values.forEach(item => this.supportedLanguageSet.add(item))
-        keys.forEach(item => this.supportedLanguageSet.add(item))
+    /**
+     * Only used when invoking CodeWhisperer service API, for telemetry usage please use mapToCodewhispererLanguage
+     */
+    public mapToCodeWhispererRuntimeLanguage(language: CodewhispererLanguage): CodewhispererLanguage {
+        switch (language) {
+            case 'jsx':
+                return 'javascript'
+
+            case 'tsx':
+                return 'typescript'
+
+            default:
+                if (!codewhispererRuntimeLanguages.has(language)) {
+                    getLogger().error(`codewhisperer: unknown runtime language ${language}`)
+                }
+                return language
+        }
     }
 
     /**
-     *
-     * @param vscLanguageId : official vscode languageId
+     * To add a new platform language id:
+     * 1. add new platform language ID constant in the file codewhisperer/constant.ts
+     * 2. add corresponding CodeWhispererLanguageId mapping in the constructor of RuntimeLanguageContext
+     * @param languageId : arbitrary string denoting a specific programming language, e.g. CodewhispererLanguageId or PlatformLanguageId
      * @returns corresponding CodewhispererLanguage ID if any, otherwise undefined
      */
-    public mapVscLanguageToCodeWhispererLanguage(vscLanguageId?: string): CodewhispererLanguage | undefined {
-        return this.supportedLanguageMap.get(vscLanguageId) ?? undefined
+    public mapToCodewhispererLanguage(languageId?: string): CodewhispererLanguage | undefined {
```

Comment:
is this only for telemetry, or is telemetry one of many uses for this function? 

the names `mapToCodeWhispererRuntimeLanguage` and `mapToCodewhispererLanguage` have no meaningful distinction. 

if this is only for telemetry consider naming it `toTelemetryLanguage`
```suggestion
    public toTelemetryLanguage(languageId?: string): CodewhispererLanguage | undefined {
```